packages/eslint-plugin/src/rules/no-unnecessary-type-constraint.ts

```diff
@@ -33,8 +33,9 @@ export default util.createRule({
   meta: {
     docs: {
       description: 'Disallows unnecessary constraints on generic types',
-      recommended: false,
+      recommended: 'error',
       suggestion: true,
+      requiresTypeChecking: true,
```

Comment:
The table in #3746 says that it needs typeChecking, but according to the tests it doesn't.

Reply:
oops - yeah it doesn't need type information - I think I copy pasted the table wrong